code/headers/moisture_sensor.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <base_module.hpp>
+#include <hwlib.hpp>
+
+namespace r2d2::plant_rescue_system {
+
+    /**
+     * Class for the moisture sensor
+     * https://nl.aliexpress.com/item/32886868425.html.
+     */
+    class moisture_sensor_c {
+        // Aanlog pin for the sensor
+        hwlib::adc &sensor_pin;
+        // The max value that the sensor can return in water.  The values are
+        // checked with a 10 bit adc of the arduino due.
+        constexpr static uint16_t max_value = 3504;
```

Comment:
Is the max value specified in the datasheet? or did you find this by testing?

Reply:
testing